core/sql/langman/LmRoutineJavaObj.cpp

```diff
@@ -192,6 +192,7 @@ LmResult LmRoutineJavaObj::invokeRoutineMethod(
           (*emitRowPtr_)(NULL,0,&qs);
         }
     }
+  jni->DeleteLocalRef(jniResult);
```

Comment:
Thanks, @sureshsubbiah, that makes sense to me. My conservative rule is the following: Assume that local refs must be explicitly deleted, don't assume that they are valid after the next (C++ to Java) JNI call.